### DISCUSSION TITLE
GPU output is all zeros

### DISCUSSION THREAD

**Discussion by mfonken-trimble** [openvinotoolkit/openvino#27652] [Q&A]
I'm having issues with my trained MobileNet SSD v2 model via OpenVINO on a Windows laptop with an Intel UHD Graphics GPU. Everything works great works great on **CPU** but returns only zeros with **GPU**. No crash or anything, just zeros on same input and just changing the **device** when running **compile_model**.

OpenVINO finds the GPU no problem with **core.available_devices**. My model is converted to OpenVINO IR format with .xml and .bin. I'm a little over my head but know the model has dynamic sizes, however I've tried reshaping the model with model.reshape([1, 640, 640, 3]) with no success. The benchmark app runs without issue on both CPU and GPU, oddly enough though the GPU is significantly slower than CPU (14FPS vs 72FPS).

Any ideas how to get GPU working? Thanks!

**Comment by Ganesh-403** [comment]
@mfonken-trimble The fact that:

- ✅ the model works on **CPU**,
- ✅ `core.available_devices` includes **GPU**,
- ✅ `benchmark_app` can run the model on the GPU,

suggests that the GPU plugin itself is working. That makes me think this is less likely to be a driver installation issue and more likely to be related to the model or how it's being executed.

A few things I'd check:

1. **Inspect the model input/output shapes**

   Even if the original model is dynamic, the GPU plugin has more limitations around dynamic shapes than the CPU plugin. Could you share the output of:

   ```python
   import openvino as ov

   core = ov.Core()
   model = core.read_model("model.xml")

   print(model.inputs)
   print(model.outputs)
   ```

   The GPU documentation notes that dynamic shape support is still limited and not all models/operations are fully supported. :contentReference[oaicite:0]{index=0}

2. **Compare CPU and GPU outputs layer by layer**

   If possible, run inference on the same input with both devices and compare the raw output tensors (before any post-processing). That will help determine whether the zeros are coming directly from the network or from downstream decoding/NMS.

3. **Try disabling Winograd convolutions**

   The GPU plugin documentation mentions that fine-tuned models can sometimes produce incorrect results on GPU when Winograd convolutions are selected. As a test, try:

   ```python
   compiled_model = core.compile_model(
       model,
       "GPU",
       config={"GPU_DISABLE_WINOGRAD_CONVOLUTION": "YES"}
   )
   ```

   and see whether the output changes. :contentReference[oaicite:1]{index=1}

4. **OpenVINO version**

   Which OpenVINO version are you using? There have been GPU plugin improvements across releases, so it's worth confirming whether this is reproducible on the latest release.

One question that would also help:

- Is your MobileNet SSD v2 a standard Open Model Zoo model, or a **custom fine-tuned TensorFlow/PyTorch model** that you converted to OpenVINO IR?

If it's a custom model, knowing the conversion path (TensorFlow → ONNX → IR, direct conversion, etc.) would help determine whether this is a conversion issue or a GPU execution issue.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
